### PR TITLE
fix(react-native-storybook): Set Storybook as a separate Activity / View Controller

### PR DESCRIPTION
**Pull request by michalziolkowski**
This PR introduces a breaking change. In non-standalone mode Storybook will be displayed with a separate Activity / View Controller and won't rely on the app to render it as a part of it's react tree. 

`getAppWithStorybook` solution gets replaced with `registerStorybook` and native module is responsible for displaying registered Storybook component on user demand or during testing on sherlo servers

Additionally, this PR refactors native modules, adding error handling and commentary on the code and removes lots of unused code in the entire package

### CODE REVIEW

File: examples/expo-example/App.tsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ import {
 } from '@expo-google-fonts/urbanist';
 import * as SplashScreen from 'expo-splash-screen';
 import { getAppWithStorybook, openStorybook } from '@sherlo/react-native-storybook';
+import Storybook from './.storybook';
+AppRegistry.registerComponent('Storybook', () => Storybook);
```
Comment (dawidk92) [inline examples/expo-example/App.tsx]: SherloStorybook

Comment (dawidk92) [reply examples/expo-example/App.tsx]: -> registerStorybook(Storybook)

File: packages/react-native-storybook/android/src/main/java/io/sherlo/storybookreactnative/SherloModule.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        final Map<String, Object> constants = new HashMap<>();
+
+        constants.put("syncDirectoryPath", this.syncDirectoryPath);
+        constants.put("initialMode", this.initialMode);
+
+        return constants;
+    }
+
+    @ReactMethod
+    public void registered(Promise promise) {
```
Comment (michalziolkowski) [inline packages/react-native-storybook/android/src/main/java/io/sherlo/storybookreactnative/SherloModule.java]: storybookRegistered

File: packages/react-native-storybook/android/src/main/java/io/sherlo/storybookreactnative/SherloModule.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        try {
+            this.isStorybookRegistered = true;
+
+            if (this.initialMode == "testing") {
+                Intent intent = new Intent(this.reactContext, SherloStorybookActivity.class);
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (michalziolkowski) [inline packages/react-native-storybook/android/src/main/java/io/sherlo/storybookreactnative/SherloModule.java:89]: komentarz co robia flagi

File: packages/react-native-storybook/ios/SherloModule.m [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+#if __has_include(<React/RCTUIManagerUtils.h>)
+// These are necessary for any interactions with the React Native UI Manager 
+// (e.g., operations on RCTRootView).
+#import <React/RCTUIManagerUtils.h>
+#endif
+
+static NSString *CONFIG_FILENAME = @"config.sherlo";
+static NSString *const LOG_TAG = @"SherloModule";
+
+static NSString *sherloDirectoryPath = @"";
```
Comment (michalziolkowski) [inline packages/react-native-storybook/ios/SherloModule.m]: syncDirectoryPath

File: packages/react-native-storybook/ios/SherloModule.m [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+RCT_EXPORT_METHOD(openStorybook:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  if(!isStorybookRegistered) {
+    return reject(@"NOT_REGISTERED", @"Storybook is not registered", nil);
+  }
+
+  [self openStorybookInternal:resolve rejecter:reject];
+}
+
+// Internal method to handle opening the Storybook view. This method is executed on the main queue.
+// If any errors occur, they are caught and the reject block is called with the appropriate error message.
```
Comment (michalziolkowski) [inline packages/react-native-storybook/ios/SherloModule.m]: komentarz ze odpala sie to na dodatowym view controllerze z ktorego mozna nawigowac spowrotem

File: packages/react-native-storybook/src/index.ts [hunk 6]
```diff
@@ -1,6 +1,5 @@
 export { default as getStorybook } from './getStorybook';
-export { default as getAppWithStorybook } from './getAppWithStorybook';
 export { default as openStorybook } from './openStorybook';
+export { default as registerStorybook } from './registerStorybook';
 
 // export { default as getSherloParameters } from './getSherloParameters';
```
Comment (michalziolkowski) [inline packages/react-native-storybook/src/index.ts]: usunac

**Review by dawidk92 — APPROVED** [review]

**Pull request merged**